ui/src/app/pages/workspace/workspace-library.tsx

```diff
@@ -67,55 +64,78 @@ const LibraryTab: React.FunctionComponent<{
     </Clickable>;
   };
 
+class ProfileState {
```

Comment:
(Recommended, but open to pushback)

I see what you're doing here (and applaud the effort to factor out a real data definition!). But a few notes / thoughts:

- It's a bit confusing to have a data type which ends up in React props, but has a class suffix of "State". Some naming ideas:
  - ReloadableProfile
  - ProfilewithReload
  - ProfileProps
  - ProfileHook
  - ProfileWrapper
- It looks like we use the same data structure in a number of files (I searched "profileState" and found a bunch of results). Should we define the interface in a more central location and reduce our overall duplication? If you opt for this, I'd suggest splitting it out into a separate tech-debt ticket and PR.
- I think we have a pretty strong pattern of using interfaces, rather than classes, to define props & state data types.